android/modules/media/src/java/ti/modules/titanium/media/TiCameraXActivity.java

```diff
@@ -0,0 +1,570 @@
+						TiBlob blob = TiBlob.blobFromFile(TiFileFactory.createTitaniumFile(path, false));
+						KrollDict response = MediaModule.createDictForImage(blob, blob.getMimeType());
+						successCallback.callAsync(callbackContext, response);
+					}
+				} else {
+					// normal file
+					File file = new File(uri.getPath());
+					TiBlob blob = TiBlob.blobFromFile(TiFileFactory.createTitaniumFile(file.getPath(), false));
+					KrollDict response = MediaModule.createDictForImage(blob, blob.getMimeType());
+					successCallback.callAsync(callbackContext, response);
+				}
+
+				if (cameraActivity != null && autohide) {
+					if (recording != null) {
+						recording.close();
+						recording = null;
+					}
+					cameraActivity.finish();
+				}
+			}
+			// All events, including VideoRecordEvent.Status, contain RecordingStats.
+			// This can be used to update the UI or track the recording duration.
+			RecordingStats recordingStats = videoRecordEvent.getRecordingStats();
+			long recordingTime = (recordingStats.getRecordedDurationNanos() / 1000 / 1000);
+			if (isRecording && videoMaximumDuration > 0 && recordingTime >= videoMaximumDuration) {
+				stopVideoCapture();
+			}
+			if (isRecording && recordingCallback != null) {
+				recordingDict.put("duration", recordingTime);
+				recordingDict.put("size", recordingStats.getNumBytesRecorded());
+				recordingCallback.callAsync(callbackContext, recordingDict);
+			}
+		};
+
+		if (cameraActivity.hasAudioRecorderPermissions()) {
+			recording = pendingRecording.withAudioEnabled().start(executor, vre);
+			isRecording = true;
+		} else {
+			recording = pendingRecording.start(executor, vre);
+			isRecording = true;
+		}
+	}
+
+	public static void pauseVideoCapture()
+	{
+		if (recording != null) recording.pause();
+		isRecording = false;
+	}
+
+	public static void resumeVideoCapture()
+	{
+		if (recording != null) recording.resume();
+		isRecording = true;
+	}
+
+	public static void stopVideoCapture()
+	{
+		if (recording != null) recording.stop();
+		isRecording = false;
+	}
+
+	public static void setFlashMode(int cameraFlashMode)
+	{
+	}
+
+	public static void hide()
+	{
+		if (recording != null) {
+			recording.close();
+			recording = null;
+		}
+		if (cameraActivity != null) {
+			cameraActivity.finish();
+		}
+	}
+
+	private static String createExternalMediaName()
+	{
+		TiApplication app = TiApplication.getInstance();
+		String normalizedAppName = app.getAppInfo().getName();
+		normalizedAppName = normalizedAppName.replaceAll("[^\\w.-]", "_");
+		return normalizedAppName + "_" + (new SimpleDateFormat("yyyyMMdd_HHmmssSSS")).format(new Date());
+	}
+
+	private static String getStringFrom(Cursor cursor, int columnIndex)
+	{
+		String result = null;
+		if ((cursor != null) && (columnIndex >= 0)) {
+			try {
+				if (cursor.getType(columnIndex) == Cursor.FIELD_TYPE_STRING) {
+					result = cursor.getString(columnIndex);
+				}
+			} catch (Exception ex) {
+			}
```

Comment:
Handle exception

Reply:
It's from https://github.com/tidev/titanium_mobile/blob/81d3f4a1bfa69a5367b8cf05af40b608b62ebe3b/android/titanium/src/java/org/appcelerator/titanium/io/TiContentFile.java#L427-L439
it will return `null` when the try-catch fails